src/SensioLabs/Melody/Handler/AuthenticableHandlerInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace SensioLabs\Melody\Handler;
+
+use SensioLabs\Melody\Configuration\UserConfiguration;
+use SensioLabs\Melody\Exception\InvalidCredentialsException;
+
+/**
+ * Interface AuthenticableHandlerInterface.
+ *
+ * @author Maxime STEINHAUSSER <maxime.steinhausser@gmail.com>
+ */
+interface AuthenticableHandlerInterface extends ResourceHandlerInterface
+{
+    const CREDENTIALS_NORMAL = 'normal';
+    const CREDENTIALS_SECRET = 'secret';
+
+    /**
+     * Returns an array of the required credentials.
+     *
+     * @return array
+     */
+    public function getRequiredCredentials();
+
+    /**
+     * Provides credentials to be used at some point by the handler.
+     *
+     * @param array             $credentials
+     * @param UserConfiguration $userConfig
+     *
+     * @throws InvalidCredentialsException
+     */
+    public function provideCredentials(array $credentials, UserConfiguration $userConfig);
```

Comment:
That's one of the aspect I'm not satisfied indeed but:
> 1. setCredentials.

It's doing much more than simply setting given credentials. (in the `GistHandler` case, using given credentials to create and save a new oauth token)

> it's not the responsibility of the AuthenticableHandlerInterface to implement the save of credentials in the user config

Yup, I agree, but then: 

1. the method has to return the token to be saved
1. I don't know better how should be named a such method :/
1. we need to expose another method like `getHandlerKey` in the interface or guess it from the handler classname.